pkg/plugins/client_store.go

```diff
@@ -0,0 +1,59 @@
+package plugins
+
+import (
+	"sync"
+
+	"github.com/hashicorp/go-plugin"
+)
+
+type clientProps struct {
+	ID   PluginIdentity
+	Auth PluginHandshake
+	P    plugin.Plugin
+}
+
+type Client struct {
+	PluginClient *plugin.Client
+	ClientProps  *clientProps
+}
+
+// clientStore maps the available plugin names to the different kinds of plugin.Client plugin controllers and their properties.
+//
+// The map might include a mapping of "delta" -> `{ plugin.Client, "deltaPluginLocation", ["arg1"], ["env1"] }` to
+// communicate with the Delta plugin.
+type clientStore struct {
```

Comment:
Can we use [sync.Map](https://pkg.go.dev/sync#Map) instead?

Reply:
Thought about doing that, but then read their documentation:

> The Map type is specialized. Most code should use a plain Go map instead, with separate locking or coordination, for better type safety and to make it easier to maintain other invariants along with the map content.

WDYT?